core/src/mindustry/ui/fragments/MenuFragment.java

```diff
@@ -296,7 +305,7 @@ public static class MenuButton{
         /** Runnable ran when the button is clicked. Ignored on desktop if {@link #submenu} is not null. */
         public final Runnable runnable;
         /** Submenu shown when this button is clicked. Used instead of {@link #runnable} on desktop. */
-        public final @Nullable MenuButton[] submenu;
```

Comment:
Not sure how I feel about a signature change here, this might crash mods on startup.

Reply:
I don't see a mods needing to access submenu directly since adding submenu buttons is already handled by the constructor.
I think it'll be fine.